packages/perseus/src/widgets/interactive-graphs/mafs-state-to-interactive-graph.ts

```diff
@@ -0,0 +1,55 @@
+import {getRadius} from "./reducer/interactive-graph-state";
+
+import type {InteractiveGraphState} from "./types";
+import type {PerseusGraphType} from "@khanacademy/perseus";
+import {UnreachableCaseError} from "@khanacademy/wonder-stuff-core";
+
+// Converts the state of a StatefulMafsGraph back to the format used to
+// represent graph state in the widget JSON.
+//
+// Rather than be tightly bound to how data was structured in
+// the legacy interactive graph, this lets us store state
+// however we want and we just transform it before handing it off
+// the the parent InteractiveGraph.
+//
+// The transformed state is used in the interactive graph widget editor, to
+// set the `correct` field of the graph options.
```

Comment:
@jeremywiebe @nishasy I'd especially like your feedback on this comment. Is this correct? Does anything besides the editor handle widgets' `onChange` event?

I found some dead test prep code in webapp that was using `onChange`, but that doesn't seem worth mentioning here since I believe it's about to be deleted. I also had a vague memory that we saved learners' progress to the backend on change, but that didn't seem to be the case when I tested - perhaps that feature was only in test prep.

Reply:
The `renderer` itself sends an `onChange` callback to widgets where it updates its state (widget state). 

I also think that some widgets that accept user input send an `onChange` event with updated prop values and expect to be re-rendered with these new values. For example, the `label-image` widget does this ([Github](https://github.com/Khan/perseus/blob/a6ec402c0e41dae1dbd980106265929ebe761bce/packages/perseus/src/widgets/label-image/label-image.tsx#L432)).

Reply:
So I don't think the editor is the only thing that uses `onChange`. I am not sure as I've never dug through, but I've long wondered if we need such an all-encompassing `onChange` handler type or if we could break it apart by widget.